daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -1070,6 +1078,25 @@ private[lf] final case class Compiler(
     }
   }
 
+  // Modify/extend `remaps` to reflect when new values are pushed on the stack. For SELet and SECase bindings
+  def shift(remaps: Map[Int, SELoc], n: Int): Map[Int, SELoc] = {
```

Comment:
?

Reply:
Yes. previously the re-mapping was performed in a context of absolute stack offsets.
This was *really really* hard to understand, although perhaps more efficient (at compile time).

It would be possible to do this still, at the cost of some confusion.

But a further problem is that the (SELocS) values of the map are also relative indexed.
And so we either need a new type to represent SELoc-S(absolute)+A+F
Or abuse the existing SELoc type.
Neither choice seems very nice.

Actually, I would prefer if the SELocS could be actually be indexed absolutely.
I think this would be slightly more efficient at runtime.
I did have an experiment, but it runs in to issues regarding the way we handle the recursive builtins.
So I thought it best to leave this potential change to a later time.